charts/athens-proxy/templates/deployment.yaml

```diff
@@ -9,6 +9,13 @@ metadata:
     heritage: "{{ .Release.Service }}"
 spec:
   replicas: {{ .Values.replicaCount }}
+{{- if .Values.strategy }}
+  strategy:
+{{ toYaml .Values.strategy | indent 4 }}
+  {{- if eq .Values.strategy.type "Recreate" }}
```

Comment:
This clause seems odd. I assume it's a bit of defensive programming, but if a user is changing the deployment strategy they should probably not be specifying values that aren't compatible.

Reply:
@twexler 

[this is the reason why I did that](https://github.com/helm/charts/commit/663bd1f7484877f4fcfa7ae9a9139cb198d7e659)

if you're still uncomfortable with that, I can remove it

Reply:
Ah, okay. This is a common pattern, interesting.